src/contracts/periphery/OperatorFactory.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import "./Operator.sol";
+import "@openzeppelin/contracts/utils/Create2.sol";
+
+contract OperatorFactory {
+    function createOperator() external returns (Operator) {
+        return Operator(
+            Create2.deploy(
+                0,
+                bytes32(uint256(uint160(msg.sender))),
+                type(Operator).creationCode
+            )
+        );
+    }
+
+    function getOperatorContract(
+        address owner
+    ) external view returns (Operator) {
+        Operator operator = Operator(Create2.computeAddress(
+            bytes32(uint256(uint160(owner))),
+            keccak256(type(Operator).creationCode)
+        ));
+    }
+}
```

Comment:
This seems like it could be pure - it also seems like without storage, if the operator code changes or upgrades or is modified for any reason this won't work.